### ISSUE TITLE
Documentation of the website

### ISSUE THREAD

**Issue by shubhank-saxena** [developer-student-club-thapar/officialWebsite#25]
This is an open source project.
So we need to write a documentation explaining the following -

1. Entire structure of the website
2. Technologies used
3. How to fork and develop (so that this website/project can be used by other communities). Make sure they do give due credits.
4. How to deploy

**Comment by animesh-007** [comment]
@shubhank-saxena  I guess this issue is now solved by updated readme.md Only thing missing in readme is technologies used.

**Comment by priyanshsaxena24** [comment]
Hey is this issue still open? @jaskeerat789 would love to contribute.

**Comment by jaskeerat789** [comment]
@deathzombie yes you can start working but make sure you register for hacktoberfest [here](http://dsctiet.xyz/hacktoberfest) since the issue is tagged as `hacktoberfest`

**Comment by jaskeerat789** [comment]
@shubhank-saxena @jsparmani Please provide the task which are unfinished in documentation. make a checklist so that we can keep adding and checking stuff accordingly.

**Comment by jsparmani** [comment]
The documentation must have these topics:
- [ ] Full Repository structure, explaining the hierarchy if folders and clearly distinguishing on frontend and backend code
- [ ] Details on Tech Stack, Libraries Used with links to their documentation. 
- [ ] Insight on How to Setup the Repo, already there in `CONTRIBUTING.md`, but must be mentioned distinctively.
- [ ] Guidelines on Pre-Commit and Travis Builds (should have info about each test being carried by travis)
- [ ] Fix to some common error messages and known bugs
- [ ] Info about the Deployment Setup (Technologies used + procedure followed)
- [ ] Some common checklists/rules to take care of before committing or opening a PR

@jaskeerat789 @deathzombie I think following points are expected to be put up in the documentation. It can be good if we can have docs in an interactive format hosted at something like `docs.dsctiet.tech` or something, just like docs of some libraries or projects.

@shubhank-saxena Please suggest if there is any discrepancy in above list or if it misses some points.

**Comment by Subhamjain02** [comment]
Hey I would like to contribute

**Comment by jaskeerat789** [comment]
Feel free to work on it and open a PR @Subhamjain02
